### ISSUE TITLE
Support 'tf' axis ordering in keras.preprocessing.image

### ISSUE THREAD

**Issue by dansbecker** [fchollet/keras#2286]
The image preprocessing code currently has the 'th' axis ordering hard-coded.  For example, the random shift function has the following

    if wrg:
        shift_x = np.random.uniform(-wrg, wrg) * x.shape[2]
    if hrg:
        shift_y = np.random.uniform(-hrg, hrg) * x.shape[1]

If the data is in 'tf' format (channels last) than the shift_x is shifting the channels rather than the width.

The same issue arises in random_rotation, horizontal_flip and vertical_flip. 

I will submit a PR addressing this.

**Comment by fchollet** [comment]
> I will submit a PR addressing this.

Cool, counting on you.
